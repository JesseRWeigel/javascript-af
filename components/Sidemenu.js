import React, { Component } from 'react'
import Sidebar, { NavIcon } from './styles/Sidebar'
import HomeIcon from '../assets/icons/home'
import StarIcon from '../assets/icons/star'
import SearchIcon from '../assets/icons/search'
import TrendingIcon from '../assets/icons/trending'
import AccountIcon from '../assets/icons/account'
import styled from 'styled-components'
import { Consumer } from './withAuth'

const Logo = styled.div`
  display: flex;
  justify-content: center;
  padding: 20% 0 20% 0;
  & > img {
    height: 40px;
    width: 40px;
  }
  @media all and (max-width: 570px) {
    display: none;
  }
`

class Sidemenu extends Component {
  render() {
    return (
      <Sidebar>
        <Logo>
          <img src="/static/logo.png" alt="" />
        </Logo>
        <NavIcon href="/">
          <HomeIcon />
        </NavIcon>
        <NavIcon href="/social">
          <TrendingIcon />
        </NavIcon>
        <NavIcon href="/starred">
          <StarIcon />
        </NavIcon>
        <NavIcon href="/search">
          <SearchIcon />
        </NavIcon>
        <Consumer>
          {user => (
            <NavIcon as={`/user/${user._id}`} href={`/user?id=${user._id}`}>
              <AccountIcon />
            </NavIcon>
          )}
        </Consumer>
      </Sidebar>
    )
  }
}

export default Sidemenu
