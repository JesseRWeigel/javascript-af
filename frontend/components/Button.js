import styled from 'styled-components'

const styles = `
  outline: none;
  color: #fff;
  font-family: 'Quicksand', Segoe UI, Tahoma, Geneva, sans-serif;
  font-weight: 600;
  font-size: 18px;
  padding: 0.8rem;
  border-radius: 50px;
  border: none;
  cursor: pointer;
  
`

const Button = styled.button`
  ${styles};
  background-color: ${props => props.theme.primary};
  &:disabled {
    cursor: not-allowed;
    background-color: ${props => props.theme.secondary};
  }
`
export default Button

export const LinkBtn = styled.a`
  ${styles};
  background-color: ${props => props.theme.primary};
  text-decoration: none;
`
export const InvertedButton = styled.a`
  outline: none;
  color: ${props => props.theme.primaryDark};
  font-family: 'Quicksand', Segoe UI, Tahoma, Geneva, sans-serif;
  font-weight: 600;
  font-size: 18px;
  transition: 0.25s ease all;
  padding: 0.8rem;
  border-radius: 50px;
  cursor: pointer;
  background-color: white;
  border: 2px solid ${props => props.theme.primaryDark};
  text-decoration: none;
  &:hover {
    color: white;
    background-color: ${props => props.theme.primary};
  }
`
