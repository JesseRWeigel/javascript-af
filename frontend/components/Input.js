import React from 'react'
import styled, { css } from 'styled-components'

const Input = styled.input`
  border: 1px solid #dfdfdf;
  font-size: 1rem;
  padding: 0.8rem;
  color: #444;
  border-radius: 3px;
  font-family: Quicksand, Roboto, sans-serif;
  font-weight: bold;
  transition: all 0.2s ease-in-out;
  outline: 0;
  display: inline-block;
  margin-bottom: 1rem;
  width: ${props => (props.full ? '90%' : 'auto')};
  ${props =>
    props.disabled
      ? css`
          cursor: not-allowed;
        `
      : ''} text-align: left;
  &:focus {
    border-color: ${props => props.theme.primary};
  }
`

const InputElement = props => {
  return <Input {...props} />
}

export const InputGroup = styled.div`
  display: flex;
  flex-direction: column;
  label {
    font-size: 20px;
    margin-bottom: 0.4rem;
    font-family: 'Quicksand', sans-serif;
  }
`

export default InputElement
