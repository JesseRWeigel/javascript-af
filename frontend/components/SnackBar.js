import React, { Component } from 'react'
import styled from 'styled-components'

export const Container = styled.div`
  box-sizing: border-box;
  display: flex;
  position: fixed;
  top: 13%;
  align-items: flex-end;
  right: 2%;
  width: 100vw;
  z-index: 1000;
  flex-direction: column;
`

const Toast = styled.div`
  background-color: #323232;
  margin-bottom: 10px;
  transition: all 0.4s ease-in-out;
  border-radius: 2px;
  display: ${props => (props.show ? 'flex' : 'none')};
  justify-content: space-between;
  max-width: 568px;
  min-width: 288px;
  line-height: 1.5;
  padding: 12px 24px;
`

const Message = styled.p`
  color: #f1f1f1;
`

const Action = styled.a`
  color: #a28aff;
  cursor: pointer;
  margin-left: 24px;
  text-transform: uppercase;
`

class SnackBar extends Component {
  state = {
    show: false
  }

  componentDidMount() {
    this.setState({ show: true }, () => {
      if (this.props.actionText) return
      setTimeout(() => {
        this.setState({ show: false })
      }, this.props.timeout || 4000)
    })
  }

  clickHandler = () => {
    this.props.actionClick && this.props.actionClick()
    this.setState({ show: false })
  }

  render() {
    const { message, actionText } = this.props
    const { show } = this.state
    return (
      <Toast show={show}>
        <Message>{message}</Message>
        {actionText && <Action onClick={this.clickHandler}>{actionText}</Action>}
      </Toast>
    )
  }
}

export default SnackBar
