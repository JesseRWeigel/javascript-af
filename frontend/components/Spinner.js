import React from 'react'
import styled from 'styled-components'

const Arc = styled.div`
  /*!
 * Load Awesome v1.1.0 (http://github.danielcardoso.net/load-awesome/)
 * Copyright 2015 Daniel Cardoso <@DanielCardoso>
 * Licensed under MIT
 */
  &,
  & > div {
    position: relative;
    -webkit-box-sizing: border-box;
    -moz-box-sizing: border-box;
    box-sizing: border-box;
  }
  & {
    display: block;
    font-size: 0;
    color: ${props => props.theme.primary};
    display: ${props => (props.hidden ? 'none' : 'block')};
  }

  & > div {
    display: inline-block;
    float: none;
    background-color: currentColor;
    border: 0 solid currentColor;
  }
  & {
    width: 32px;
    height: 32px;
  }
  & > div {
    position: absolute;
    top: 50%;
    left: 50%;
    border-radius: 100%;
  }
  & > div:first-child {
    position: absolute;
    width: 32px;
    height: 32px;
    background: transparent;
    border-style: solid;
    border-width: 2px;
    border-right-color: transparent;
    border-left-color: transparent;
    -webkit-animation: ball-clip-rotate-pulse-rotate 1s cubic-bezier(0.09, 0.57, 0.49, 0.9) infinite;
    -moz-animation: ball-clip-rotate-pulse-rotate 1s cubic-bezier(0.09, 0.57, 0.49, 0.9) infinite;
    -o-animation: ball-clip-rotate-pulse-rotate 1s cubic-bezier(0.09, 0.57, 0.49, 0.9) infinite;
    animation: ball-clip-rotate-pulse-rotate 1s cubic-bezier(0.09, 0.57, 0.49, 0.9) infinite;
  }
  & > div:last-child {
    width: 16px;
    height: 16px;
    -webkit-animation: ball-clip-rotate-pulse-scale 1s cubic-bezier(0.09, 0.57, 0.49, 0.9) infinite;
    -moz-animation: ball-clip-rotate-pulse-scale 1s cubic-bezier(0.09, 0.57, 0.49, 0.9) infinite;
    -o-animation: ball-clip-rotate-pulse-scale 1s cubic-bezier(0.09, 0.57, 0.49, 0.9) infinite;
    animation: ball-clip-rotate-pulse-scale 1s cubic-bezier(0.09, 0.57, 0.49, 0.9) infinite;
  }
  &.la-sm {
    width: 16px;
    height: 16px;
  }
  &.la-sm > div:first-child {
    width: 16px;
    height: 16px;
    border-width: 1px;
  }
  &.la-sm > div:last-child {
    width: 8px;
    height: 8px;
  }
  &.la-2x {
    width: 64px;
    height: 64px;
  }
  &.la-2x > div:first-child {
    width: 64px;
    height: 64px;
    border-width: 4px;
  }
  &.la-2x > div:last-child {
    width: 32px;
    height: 32px;
  }
  &.la-3x {
    width: 96px;
    height: 96px;
  }
  &.la-3x > div:first-child {
    width: 96px;
    height: 96px;
    border-width: 6px;
  }
  &.la-3x > div:last-child {
    width: 48px;
    height: 48px;
  }
  /*
 * Animations
 */
  @-webkit-keyframes ball-clip-rotate-pulse-rotate {
    0% {
      -webkit-transform: translate(-50%, -50%) rotate(0deg);
      transform: translate(-50%, -50%) rotate(0deg);
    }
    50% {
      -webkit-transform: translate(-50%, -50%) rotate(180deg);
      transform: translate(-50%, -50%) rotate(180deg);
    }
    100% {
      -webkit-transform: translate(-50%, -50%) rotate(360deg);
      transform: translate(-50%, -50%) rotate(360deg);
    }
  }
  @-moz-keyframes ball-clip-rotate-pulse-rotate {
    0% {
      -moz-transform: translate(-50%, -50%) rotate(0deg);
      transform: translate(-50%, -50%) rotate(0deg);
    }
    50% {
      -moz-transform: translate(-50%, -50%) rotate(180deg);
      transform: translate(-50%, -50%) rotate(180deg);
    }
    100% {
      -moz-transform: translate(-50%, -50%) rotate(360deg);
      transform: translate(-50%, -50%) rotate(360deg);
    }
  }
  @-o-keyframes ball-clip-rotate-pulse-rotate {
    0% {
      -o-transform: translate(-50%, -50%) rotate(0deg);
      transform: translate(-50%, -50%) rotate(0deg);
    }
    50% {
      -o-transform: translate(-50%, -50%) rotate(180deg);
      transform: translate(-50%, -50%) rotate(180deg);
    }
    100% {
      -o-transform: translate(-50%, -50%) rotate(360deg);
      transform: translate(-50%, -50%) rotate(360deg);
    }
  }
  @keyframes ball-clip-rotate-pulse-rotate {
    0% {
      -webkit-transform: translate(-50%, -50%) rotate(0deg);
      -moz-transform: translate(-50%, -50%) rotate(0deg);
      -o-transform: translate(-50%, -50%) rotate(0deg);
      transform: translate(-50%, -50%) rotate(0deg);
    }
    50% {
      -webkit-transform: translate(-50%, -50%) rotate(180deg);
      -moz-transform: translate(-50%, -50%) rotate(180deg);
      -o-transform: translate(-50%, -50%) rotate(180deg);
      transform: translate(-50%, -50%) rotate(180deg);
    }
    100% {
      -webkit-transform: translate(-50%, -50%) rotate(360deg);
      -moz-transform: translate(-50%, -50%) rotate(360deg);
      -o-transform: translate(-50%, -50%) rotate(360deg);
      transform: translate(-50%, -50%) rotate(360deg);
    }
  }
  @-webkit-keyframes ball-clip-rotate-pulse-scale {
    0%,
    100% {
      opacity: 1;
      -webkit-transform: translate(-50%, -50%) scale(1);
      transform: translate(-50%, -50%) scale(1);
    }
    30% {
      opacity: 0.3;
      -webkit-transform: translate(-50%, -50%) scale(0.15);
      transform: translate(-50%, -50%) scale(0.15);
    }
  }
  @-moz-keyframes ball-clip-rotate-pulse-scale {
    0%,
    100% {
      opacity: 1;
      -moz-transform: translate(-50%, -50%) scale(1);
      transform: translate(-50%, -50%) scale(1);
    }
    30% {
      opacity: 0.3;
      -moz-transform: translate(-50%, -50%) scale(0.15);
      transform: translate(-50%, -50%) scale(0.15);
    }
  }
  @-o-keyframes ball-clip-rotate-pulse-scale {
    0%,
    100% {
      opacity: 1;
      -o-transform: translate(-50%, -50%) scale(1);
      transform: translate(-50%, -50%) scale(1);
    }
    30% {
      opacity: 0.3;
      -o-transform: translate(-50%, -50%) scale(0.15);
      transform: translate(-50%, -50%) scale(0.15);
    }
  }
  @keyframes ball-clip-rotate-pulse-scale {
    0%,
    100% {
      opacity: 1;
      -webkit-transform: translate(-50%, -50%) scale(1);
      -moz-transform: translate(-50%, -50%) scale(1);
      -o-transform: translate(-50%, -50%) scale(1);
      transform: translate(-50%, -50%) scale(1);
    }
    30% {
      opacity: 0.3;
      -webkit-transform: translate(-50%, -50%) scale(0.15);
      -moz-transform: translate(-50%, -50%) scale(0.15);
      -o-transform: translate(-50%, -50%) scale(0.15);
      transform: translate(-50%, -50%) scale(0.15);
    }
  }
`

const Spinner = props => {
  return (
    <Arc hidden={props.hidden} {...props}>
      <div /> <div />
    </Arc>
  )
}

export default Spinner
