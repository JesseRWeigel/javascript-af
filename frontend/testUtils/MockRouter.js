import React from 'react'
import PropTypes from 'prop-types'

class MockRouter extends React.Component {
  getChildContext() {
    return {
      router: {
        pathname: this.props.href || '/',
        push: () => {},
        replace: () => {},
        reload: () => {},
        back: () => {},
        prefetch: () => {},
        beforePopState: () => {}
      }
    }
  }
  render() {
    return <>{this.props.children}</>
  }
}

MockRouter.childContextTypes = {
  router: PropTypes.object
}

export default MockRouter
