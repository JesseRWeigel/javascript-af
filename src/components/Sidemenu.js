import React from 'react'
import Sidebar, { NavIcon } from './styles/Sidebar'
import HomeIcon from '../assets/icons/home'
import SearchIcon from '../assets/icons/search'
import WorkIcon from '../assets/icons/work'
import ChatIcon from '../assets/icons/chat'
import AccountIcon from '../assets/icons/account'
import AssignmentIcon from '../assets/icons/assignment'
import styled from 'styled-components'

const Logo = styled.div`
  display: flex;
  justify-content: center;
  padding: 20% 0 20% 0;
  & > img {
    height: 40px;
    width: 40px;
  }
`

const Sidemenu = () => {
  return (
    <Sidebar>
      <Logo>
        <img src={require('../assets/logo.png')} alt="" />
      </Logo>
      <NavIcon activeClassName="active" to="/home">
        <HomeIcon />
      </NavIcon>
      <NavIcon activeClassName="active" to="/feed">
        <AssignmentIcon />
      </NavIcon>
      <NavIcon activeClassName="active" to="/jobs">
        <WorkIcon />
      </NavIcon>
      <NavIcon activeClassName="active" to="/search">
        <SearchIcon />
      </NavIcon>
      <NavIcon activeClassName="active" to="/social">
        <ChatIcon />
      </NavIcon>
      <NavIcon activeClassName="active" to="/profile">
        <AccountIcon />
      </NavIcon>
    </Sidebar>
  )
}

export default Sidemenu
