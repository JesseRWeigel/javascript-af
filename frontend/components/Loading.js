import React from 'react'
import Head from 'next/head'
import styled from 'styled-components'
import Spinner from './Spinner'

const Background = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
`
const Loader = () => (
  <Background>
    <Head>
      <title>Loading...</title>
    </Head>
    <Spinner className="la-3x" />
  </Background>
)

export default Loader
