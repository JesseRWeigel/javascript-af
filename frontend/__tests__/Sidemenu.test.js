import React from 'react'
import { render } from 'react-testing-library'
import Router from 'next/router'
import Sidemenu from '../components/Sidemenu'
import { Provider } from '../components/withAuth'
import MockRouter from '../testUtils/MockRouter'

// Mock the router
Router.router = {
  push() {},
  pathname: '/',
  replace: () => {},
  reload: () => {},
  back: () => {},
  prefetch: () => {},
  beforePopState: () => {}
}

jest.mock('next/config', () => {
  return () => ({
    publicRuntimeConfig: {
      BACKEND: 'https://localhost:8080'
    }
  })
})

describe('Sidemenu', () => {
  const user = {
    _id: '5af09c857eba02005a03f756',
    bio: 'I am student. I am studying in class XII . I love programming.',
    email: 'pantharshit00@gmail.com',
    name: 'Harshit Pant',
    photoURL: 'https://avatars3.githubusercontent.com/u/22195362?v=4'
  }
  function getTree(url) {
    return (
      <MockRouter href={url}>
        <Provider value={user}>
          <Sidemenu />
        </Provider>
      </MockRouter>
    )
  }
  test('it should render properly (Snapshot)', () => {
    expect(render(getTree('/')).container.firstChild).toMatchSnapshot()
  })

  test('it should have correct active class', () => {
    Router.router.pathname = '/trending'
    const { getByTestId } = render(getTree('/trending'))
    expect(getByTestId('tren').classList.contains('active')).toBeTruthy()
  })
})
