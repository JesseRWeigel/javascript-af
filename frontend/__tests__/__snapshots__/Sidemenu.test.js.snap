// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidemenu it should render properly (Snapshot) 1`] = `
<div
  class="Sidebar-s1k8e3n3-0 grVhnF"
>
  <div
    class="Sidemenu__Logo-pz2x40-0 jycxIV"
  >
    <img
      alt=""
      src="/static/logo.png"
    />
  </div>
  <a
    class="active Sidebar__NavA-s1k8e3n3-1 kMbYHo"
    href="/"
  >
    <svg
      aria-hidden="true"
      fill="#fff"
      height="40"
      version="1.1"
      viewBox="0 0 10 16"
      width="40"
    >
      <path
        d="M10 5c0-1.11-.89-2-2-2a1.993 1.993 0 0 0-1 3.72v.3c-.02.52-.23.98-.63 1.38-.4.4-.86.61-1.38.63-.83.02-1.48.16-2 .45V4.72a1.993 1.993 0 0 0-1-3.72C.88 1 0 1.89 0 3a2 2 0 0 0 1 1.72v6.56c-.59.35-1 .99-1 1.72 0 1.11.89 2 2 2 1.11 0 2-.89 2-2 0-.53-.2-1-.53-1.36.09-.06.48-.41.59-.47.25-.11.56-.17.94-.17 1.05-.05 1.95-.45 2.75-1.25S8.95 7.77 9 6.73h-.02C9.59 6.37 10 5.73 10 5zM2 1.8c.66 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2C1.35 4.2.8 3.65.8 3c0-.65.55-1.2 1.2-1.2zm0 12.41c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2zm6-8c-.66 0-1.2-.55-1.2-1.2 0-.65.55-1.2 1.2-1.2.65 0 1.2.55 1.2 1.2 0 .65-.55 1.2-1.2 1.2z"
        fill-rule="evenodd"
      />
    </svg>
  </a>
  <a
    class="Sidebar__NavA-s1k8e3n3-1 kMbYHo"
    data-testid="tren"
    href="/trending"
  >
    <svg
      fill="#FFFFFF"
      height="40"
      viewBox="0 0 24 24"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M13.5.67s.74 2.65.74 4.8c0 2.06-1.35 3.73-3.41 3.73-2.07 0-3.63-1.67-3.63-3.73l.03-.36C5.21 7.51 4 10.62 4 14c0 4.42 3.58 8 8 8s8-3.58 8-8C20 8.61 17.41 3.8 13.5.67zM11.71 19c-1.78 0-3.22-1.4-3.22-3.14 0-1.62 1.05-2.76 2.81-3.12 1.77-.36 3.6-1.21 4.62-2.58.39 1.29.59 2.65.59 4.04 0 2.65-2.15 4.8-4.8 4.8z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </a>
  <a
    class="Sidebar__NavA-s1k8e3n3-1 kMbYHo"
    href="/editors-choice"
  >
    <svg
      fill="#FFFFFF"
      height="40"
      viewBox="0 0 24 24"
      width="40"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zm4.24 16L12 15.45 7.77 18l1.12-4.81-3.73-3.23 4.92-.42L12 5l1.92 4.53 4.92.42-3.73 3.23L16.23 18z"
      />
    </svg>
  </a>
  <a
    class="Sidebar__NavA-s1k8e3n3-1 kMbYHo"
    href="/user/5af09c857eba02005a03f756"
  >
    <svg
      fill="#fff"
      height="50"
      viewBox="0 0 24 24"
      width="50"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </a>
</div>
`;
